dbt/adapters/athena/impl.py

```diff
@@ -56,59 +56,80 @@ def convert_number_type(cls, agate_table: agate.Table, col_idx: int) -> str:
-        if not lf_tags:
-            logger.debug("No LF tags configured")
-            return
-
-        resource = {
-            "Database": {"Name": database},
-        }
-
-        if table:
-            resource = {
-                "Table": {
-                    "DatabaseName": database,
-                    "Name": table,
-                }
-            }
 
-        with boto3_client_lock:
-            lf_client = client.session.client(
-                "lakeformation", region_name=client.region_name, config=get_boto3_config()
-            )
+        if not lf_tags and not lf_tags_columns:
+            logger.debug("No LF tags configured")
+        else:
+            with boto3_client_lock:
+                lf_client = client.session.client(
+                    "lakeformation", region_name=client.region_name, config=get_boto3_config()
+                )
 
-        response = lf_client.add_lf_tags_to_resource(
-            Resource=resource,
-            LFTags=[
-                {
-                    "TagKey": key,
-                    "TagValues": [
-                        value,
-                    ],
-                }
-                for key, value in lf_tags.items()
-            ],
-        )
+            if lf_tags:
+                resource = {"Database": {"Name": database}}
+                if table:
+                    resource = {"Table": {"DatabaseName": database, "Name": table}}
 
-        failures = response.get("Failures", [])
-        tbl_appendix = f".{table}" if table else ""
-        if failures:
-            base_msg = f"Failed to add LF tags: {lf_tags} to {database}" + tbl_appendix
-            for failure in failures:
-                tag = failure.get("LFTag", {}).get("TagKey")
-                error = failure.get("Error", {}).get("ErrorMessage")
-                logger.error(f"Failed to set {tag} for {database}" + tbl_appendix + f" - {error}")
-            raise DbtRuntimeError(base_msg)
-        else:
-            logger.debug(f"Added LF tags: {lf_tags} to {database}" + tbl_appendix)
+                response = lf_client.add_lf_tags_to_resource(
+                    Resource=resource, LFTags=[{"TagKey": key, "TagValues": [value]} for key, value in lf_tags.items()]
+                )
+                logger.debug(self.parse_lf_response(response, database, table, None, lf_tags))
+
+            if lf_tags_columns:
+                # Validate lf_tags_columns before start to interact with AWS API to apply tags
+                for tag_key, tag_config in lf_tags_columns.items():
+                    if isinstance(tag_config, Dict):
+                        for tag_value, columns in tag_config.items():
+                            if not isinstance(columns, List):
+                                raise DbtRuntimeError(f"Not a list: {columns}." + "Expected: ['c1', 'c2']")
+                    else:
+                        raise DbtRuntimeError(f"Not a dict: {tag_config}." + "Expected: {'tag_value': ['c1', 'c2']}")
```

Comment:
could you move that into a separated function please? and unit test it:D

Reply:
@svdimchenko did you see this last comment?

Reply:
yeah, now I see it. I thought about it yesterday, but it was too late actually :-) will do that today

Reply:
it's the last small change, I think that will keep the code more readable :)

Reply:
updated, pls review